Update Letter Case label name

SavPhill opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->


## What?
<!-- In a few words, what is the PR actually doing? -->

In the Settings Sidebar, the Typography Panel incorrectly labels the letter case where every first word in the sentance is a capital letter, as 'Capitalize'. The correct terminology is either 'Initial Caps' or 'Start Case', as Capitalize is not an official case variation.

Definition: https://www.yourdictionary.com/initial-caps

'Capitalize' case is the wrong label, as it's not an official case terminology and is a vague description of what the case is, which is a starting capital letter on every word.

Fixes #48617.

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
The label misrepresents the case.

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->
This fixes the issue by providing the correct label.
## Testing Instructions
1. Open the block editor and review the Typography in the Tools Panel. 
2. Hover over the 4 different cases available.
<img width="1357" alt="title-case-gutenberg" src="https://user-images.githubusercontent.com/38789408/222065531-2d1c7f83-4f37-45d8-8f81-29f006a1d181.png">


### Testing Instructions for Keyboard
<!-- How can you test the changes by using the keyboard only? Please note, this is required for PRs that change the user interface (UI). This ensures the PR can be tested for any possible accessibility regressions. -->

## Screenshots or screencast <!-- if applicable -->

SavPhill committed:
Update the correct terminology

ghost commented:
Fixes #48617

ghost commented:
Thanks for the PR @SavPhill 

> Fixes #48617

Apparently it does not work if I add it in a comment. Can you please add this line to your first message, [per our format](https://make.wordpress.org/core/handbook/best-practices/commit-messages/#format), so that these get linked.

<img width="345" alt="Screenshot 2023-03-01 at 1 01 01 PM" src="https://user-images.githubusercontent.com/118618474/222133958-febd70e4-94b5-4ac6-bd5f-e066b28d9499.png">

SavPhill commented:
> Thanks for the PR @SavPhill 
> 
> 
> 
> > Fixes #48617
> 
> 
> 
> Apparently it does not work if I add it in a comment. Can you please add this line to your first message, [per our format](https://make.wordpress.org/core/handbook/best-practices/commit-messages/#format), so that these get linked.
> 
> 
> 
> <img width="345" alt="Screenshot 2023-03-01 at 1 01 01 PM" src="https://user-images.githubusercontent.com/118618474/222133958-febd70e4-94b5-4ac6-bd5f-e066b28d9499.png">
> 
> 

Thanks Tom. I will follow this process in future, and have added this to my original detail as instructed in the guidelines: https://make.wordpress.org/core/handbook/best-practices/commit-messages/#ticket-references

SavPhill commented:
Bumping this reduce the amount of stagnating PR's. 

Anything else required here or can we approve/reject? @ellatrix

skorasaurus commented:
Thanks for the PR @SavPhill . 

'Start Case' would be more appropriate (https://en.wikipedia.org/wiki/Letter_case#Stylistic_or_specialised_usage) as you stated; although I'm not sure whether all of the named capitalized functions in that component should also be renamed? 

Perhaps a good first step would be just to rename the copy and have a comment in the code that it was formerly called capitalize

SavPhill committed:
Update index.js

Update Initial Caps to Start Case. Add a Note to explain the change.

SavPhill commented:
> Thanks for the PR @SavPhill .
> 
> 'Start Case' would be more appropriate (https://en.wikipedia.org/wiki/Letter_case#Stylistic_or_specialised_usage) as you stated; although I'm not sure whether all of the named capitalized functions in that component should also be renamed?
> 
> Perhaps a good first step would be just to rename the copy and have a comment in the code that it was formerly called capitalize

Thank you Will. I have updated to 'Start Case' as suggested, and added a one line comment showing that it was previously 'Captalize', to ensure there is no misunderstand. Perhaps we can look at renaming the rest of the code at a different stage.

SavPhill committed:
Merge branch 'WordPress:trunk' into patch-15

SavPhill commented:
Bump to reduce stagnation Any further feedback for this PR?

SavPhill commented:
Bumping this one again to reduce stagnation.
